kernel/src/lib.rs

```diff
@@ -196,6 +197,38 @@ pub trait JsonHandler: Send + Sync {
         // clone the (potentially large) expression every time we call this function.
         predicate: Option<Expression>,
     ) -> DeltaResult<FileDataReadResultIterator>;
+
+    /// Atomically (!) write a single JSON file. Each row of the input data represents an action
```

Comment:
Nit: We shouldn't specify the contents that are being written.

For example, `_LAST_CHECKPOINT` doesn't contain actions from the delta log.

https://github.com/delta-io/delta/blob/master/PROTOCOL.md#last-checkpoint-file-schema

Reply:
yea good point removing